prost-types/src/conversions.rs

```diff
@@ -0,0 +1,62 @@
+use crate::protobuf::Value;
+use crate::value;
+use crate::String;
+use crate::Vec;
+use ::prost::alloc::collections::BTreeMap;
+
+impl From<value::Kind> for Value {
+    fn from(value: value::Kind) -> Self {
+        Value { kind: Some(value) }
+    }
+}
+
+macro_rules! impl_number_value {
+    ($t: ty) => {
+        impl From<$t> for Value {
+            fn from(value: $t) -> Self {
+                value::Kind::NumberValue(value.into()).into()
+            }
+        }
+    };
+}
+
+impl_number_value!(u8);
+impl_number_value!(u16);
+impl_number_value!(u32);
+
+impl_number_value!(i8);
+impl_number_value!(i16);
+impl_number_value!(i32);
+
+impl_number_value!(f32);
+impl_number_value!(f64);
+
+impl From<bool> for Value {
+    fn from(value: bool) -> Self {
+        value::Kind::BoolValue(value).into()
+    }
+}
+
+impl From<String> for Value {
+    fn from(value: String) -> Self {
+        value::Kind::StringValue(value).into()
+    }
+}
+
+impl<'a> From<&'a str> for Value {
+    fn from(value: &str) -> Self {
+        value::Kind::StringValue(value.into()).into()
+    }
+}
```

Comment:
Nope

```
the trait `Into` cannot be made into an object
the trait cannot be made into an object because it requires `Self: Sized`
```

```
the size for values of type `(dyn Into<std::string::String> + 'static)` cannot be known at compilation time
```